pkg/buildchange/change_processor.go

```diff
@@ -0,0 +1,111 @@
+package buildchange
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+const (
+	reasonsSeparator = ","
+	errorSeparator   = "\n"
+)
+
+func NewChangeProcessor() *ChangeProcessor {
+	return &ChangeProcessor{
+		changes: []GenericChange{},
+		errStrs: []string{},
+	}
+}
+
+type ChangeProcessor struct {
+	changes []GenericChange
+	errStrs []string
+}
+
+func (c *ChangeProcessor) Process(change Change) *ChangeProcessor {
+	if change == nil {
+		return c
+	}
+
+	if !change.Reason().IsValid() {
```

Comment:
Is there a reason to validate this? It seems like the change processor does not need care about the change being a `known` type.

Reply:
Yes, it can be avoided since all the changes are defined by us anyway. Will take it out.

Reply:
@matthewmcnew please let me know if you still consider this as not necessary. I would prefer to keep it. Either way, would like to resolve this comment.

Reply:
I seems like a bit of an unnecessary open/closed violation for the change processor to need to know about the known types of acceptable buildReasons. I don't think we need to be concerned about internals of kpack producing a random buildReason string as this does not come from a user input.

Reply:
[This comment](https://github.com/pivotal/kpack/pull/556#discussion_r540403788) applies here too

Reply:
addressed in https://github.com/pivotal/kpack/commit/e73935fa92379241fdd65e1d2d916a4ae487afc5